packages/teams-js/src/private/store.ts

```diff
@@ -0,0 +1,140 @@
+    ICS: 'https://teams.microsoft.com/extensibility-apps/hostedincontextstore/create?language={locale}&metaoshost=office&host=metaos&clienttype=web',
+    appDetail:
+      'https://teams.microsoft.com/extensibility-apps/appdetails/{appId}/create?language={locale}&metaoshost=office&host=metaos&clienttype=web',
+    collectionStore:
+      'https://teams.microsoft.com/extensibility-apps/store/app:co:{collectionId}?language={locale}&metaoshost=office&host=metaos&clienttype=web',
+  };
+  /**
+   * @beta
+   * @hidden
+   * error message when trying to open app detail dialog but missing app id
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export const errorMissingAppId = 'Missing App Id';
+  const StoreVersionTagNum = ApiVersionNumber.V_2;
+  function getStoreUrl(openStoreParams: OpenStoreParams): string {
+    const { dialogType, appId, userHasCopilotLicense, collectionId } = openStoreParams;
+    if (dialogType === StoreDialogTypeEnum.fullStore) {
+      if (userHasCopilotLicense) {
+        return StoreUrl.collectionStore.replace('{collectionId}', 'copilotplugins');
+      }
+      if (collectionId !== undefined) {
+        return StoreUrl.collectionStore.replace('{collectionId}', collectionId);
+      }
+      return StoreUrl.fullStore;
+    }
+
+    if (dialogType === StoreDialogTypeEnum.ICS) {
+      return StoreUrl.ICS;
+    }
+
+    if (dialogType === StoreDialogTypeEnum.appDetail) {
+      if (appId === undefined) {
+        throw new Error(errorMissingAppId);
+      }
+      return StoreUrl.appDetail.replace('appId', appId);
+    }
+
+    return StoreUrl.fullStore;
+  }
+  /**
+   * @beta
+   * @hidden
+   * Api to open a store
+   *
+   * @param openStoreParams - params to call openStoreExperience
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export async function openStoreExperience(openStoreParams: OpenStoreParams): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
+    if (!this.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const context = await app.getContext();
+    const url: string = getStoreUrl(openStoreParams).replace('{locale}', context.app.locale);
+    const storeDialogInfo: UrlDialogInfo = {
+      url,
+      size: {
+        height: DialogDimension.Medium,
+        width: DialogDimension.Medium,
+      },
+    };
+    const dialogInfo: DialogInfo = dialog.url.getDialogInfoFromUrlDialogInfo(storeDialogInfo);
+    sendMessageToParent(getApiVersionTag(StoreVersionTagNum, ApiName.Store_Open), 'store.open', [dialogInfo]);
+  }
+  /**
+   * Checks if the store capability is supported by the host
+   * @returns boolean to represent whether the store capability is supported
+   *
+   * @throws Error if {@linkcode app.initialize} has not successfully completed
+   */
+  export function isSupported(): boolean {
+    return ensureInitialized(runtime);
```

Comment:
This function should be checking the runtime object to make sure that the host is correctly marking this capability as supported. Check other versions of `isSupported` for examples.

Reply:
updated